### ISSUE TITLE
kubectl run --dry-run=client does not put --namespace info in yaml 

### ISSUE THREAD

**Issue by abhinavg97** [kubernetes/kubernetes#92899]
**What happened**:

`kubectl run` command does not fill the namespace info in the yaml when operated with the `--dry-run=client -o yaml` option

**What you expected to happen**:

The yaml has the namespace info populated as specified in the run command

**How to reproduce it (as minimally and precisely as possible)**:

```
kubectl run temp --image=nginx --namespace=my-ns -o yaml --dry-run=client
```
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.18
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`): Ubuntu 18.04
- Kernel (e.g. `uname -a`):  Linux master 4.15.0-101-generic
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by abhi1604** [comment]
sig/cli

**Comment by k-toyoda-pi** [comment]
Hi. Did you mean that yaml has no namespace when you run the command?
It seems that the latest version has namespace. 
```
$ cluster/kubectl.sh run temp --image=nginx --namespace=my-ns -o yaml --dry-run=client
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  labels:
    run: temp
  name: temp
  namespace: my-ns <-- this?
spec:
  containers:
  - image: nginx
    name: temp
    resources: {}
  dnsPolicy: ClusterFirst
  restartPolicy: Always
status: {}

$ cluster/kubectl.sh version
Client Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.0-beta.2.788+bf94f27e76c541-dirty", GitCommit:"bf94f27e76c541db57098ed69aea47d889703669", GitTreeState:"dirty", BuildDate:"2020-07-09T01:29:48Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.0-beta.2.788+bf94f27e76c541-dirty", GitCommit:"bf94f27e76c541db57098ed69aea47d889703669", GitTreeState:"dirty", BuildDate:"2020-07-09T01:29:48Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by abhi1604** [comment]
> Hi. Did you mean that yaml has no namespace when you run the command?
> It seems that the latest version has namespace.
> 
> ```
> $ cluster/kubectl.sh run temp --image=nginx --namespace=my-ns -o yaml --dry-run=client
> apiVersion: v1
> kind: Pod
> metadata:
>   creationTimestamp: null
>   labels:
>     run: temp
>   name: temp
>   namespace: my-ns <-- this?
> spec:
>   containers:
>   - image: nginx
>     name: temp
>     resources: {}
>   dnsPolicy: ClusterFirst
>   restartPolicy: Always
> status: {}
> 
> $ cluster/kubectl.sh version
> Client Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.0-beta.2.788+bf94f27e76c541-dirty", GitCommit:"bf94f27e76c541db57098ed69aea47d889703669", GitTreeState:"dirty", BuildDate:"2020-07-09T01:29:48Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"linux/amd64"}
> Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.0-beta.2.788+bf94f27e76c541-dirty", GitCommit:"bf94f27e76c541db57098ed69aea47d889703669", GitTreeState:"dirty", BuildDate:"2020-07-09T01:29:48Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"linux/amd64"}
> ```

Yes, I think it got corrected in the 1.19 version. This issue is pertaining to the 1.18 version.
